spring-xd-dirt/src/main/java/org/springframework/xd/dirt/cluster/ContainerFilter.java

```diff
@@ -28,6 +28,6 @@
  */
 public interface ContainerFilter {
 
-	List<Container> filterContainers(ModuleDescriptor moduleDescriptor, Iterable<Container> containers);
+	Collection<Container> filterContainers(ModuleDescriptor moduleDescriptor, Iterable<Container> containers);
```

Comment:
I like the idea of a boolean "accept" method actually. However, that would prevent use-cases where a decision is made based on some decision taken on the group as a whole.